pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -402,6 +406,21 @@ stdenv.mkDerivation ({
         export NIX_${ghcCommandCaps}PKG="${ghcEnv}/bin/${ghcCommand}-pkg"
         # TODO: is this still valid?
         export NIX_${ghcCommandCaps}_DOCDIR="${ghcEnv}/share/doc/ghc/html"
+        export LD_LIBRARY_PATH="''${LD_LIBRARY_PATH:+''${LD_LIBRARY_PATH}:}${
```

Comment:
Setting `LD_LIBRARY_PATH` in the generic builder is not a good idea. It's important that all relevant search paths are registered with Cabal because otherwise the generated library config file does not contain them and then libraries will fail to link when they're used by other components, etc.

Reply:
What is the right way to register this with Cabal?

Reply:
Ah but this is the shell env. @peti the `env` is really screwed up, you know, because the system-haskell division of deps does not take into account propagated dependencies. We drop propagated systems deps (including frameworks) on the floor since `ghcWithPackages` doesn't reexport them. I like the idea of `ghcWithPackages`, but it is very incompatible with the bash-heavy cc-wrapper way of doing things. More thought is needed.

Reply:
I'll make a separate PR for `--extra-framework-dirs`.

Reply:
@ElvishJerricco let's try to land https://github.com/NixOS/nixpkgs/pull/39735 by @kirelagin for this reflex-platform release too.

Reply:
@Ericson2314 So you're going to tackle these `env` issues in a separate PR? Should I remove the `env` changes from this PR entirely?